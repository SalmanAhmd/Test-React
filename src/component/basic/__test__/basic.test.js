import Basic from '../';

import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';


// import TestRenderer from 'react-test-renderer';
// import ShallowRenderer from 'react-test-renderer/shallow';


// Basic Test with React-test-renderer
// it('renders correctly react-test-renderer', () => {
//   const renderer = new ShallowRenderer();
//   renderer.render(<Basic />);
//   const result = renderer.getRenderOutput();
//   expect(result).toMatchSnapshot();
// });


// Basic Test with Enzyme
it('renders correctly enzyme shallow', () => {
  const wrapper = shallow(<Basic />)
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders correctly enzyme mount', () => {
  const wrapper = mount(<Basic />)
  expect(toJson(wrapper)).toMatchSnapshot();
  wrapper.unmount();
});
