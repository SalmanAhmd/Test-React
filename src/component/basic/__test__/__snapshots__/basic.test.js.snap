// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme mount 1`] = `
<Basic>
  <div>
    <h1>
       Basic Test
    </h1>
    <p>
       This is a basic Test Component
    </p>
  </div>
</Basic>
`;

exports[`renders correctly enzyme shallow 1`] = `
<div>
  <h1>
     Basic Test
  </h1>
  <p>
     This is a basic Test Component
  </p>
</div>
`;
